capstone_project/src/main/webapp/scripts/mapsScriptTemplate.js

```diff
@@ -174,13 +174,58 @@ function createNewsFeed(articles) {
   return newsFeed;
 }
 
+/** Converts post objects into links to be featured under the "Posts" tab.
+ * @param {Array.<Object>} schoolPosts - A list of posts pertaining to a school.
+ * @param {string} schoolName - The name of a school.
+ * @return {string} - The HTML code for providing either post links or an option
+ * to make the first post for a school.
+ */
+function createLinks(schoolPosts, schoolName) {
+  // If there are no existing posts for a school, provide the user with a link
+  // to the page where they can create one.
+  if (schoolPosts === undefined|| schoolPosts.length == 0) {
+    return `<h5>
+                There are no posts for ${schoolName} yet. Click
+                <a href="../pages/comments.jsp">here</a>
+                to be the first to make one.
+            </h5>`;
+  }
+
+  // Generate an unordered list of links to individual posts.
+  let postFeed= `<h5><b>Check out posts from students at ${name}:</b></h5>`;
+  postFeed += '<ul>';
+  for (const post of schoolPosts) {
+    postFeed += `<li>
+                  <a href="../pages/post_displayer.html?post-id=${post.key.id}">
+                  ${post.propertyMap.title}</a>
+                </li>`;
+  }
+  postFeed += '</ul>';
+  return postFeed;
+}
+
+/** Fetches a school's posts and creates the "Posts" tab portion of a content
+ * string for a marker.
+ * @param {string} schoolName - The name of a school.
+ * @return {string} - A postFeed string containing the HTML code for
+ * the "Posts" tab of an infowindow.
+ */
+async function createPostsFeed(schoolName) {
+  let schoolPosts;
+  await fetch(`/fetch-school-posts?school-name=${schoolName}`)
```

Comment:
can you encode the school name with [encodeURIComponent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) since it's being passed into a url